roofit/roostats/src/MCMCInterval.cxx

```diff
@@ -296,8 +296,11 @@ void MCMCInterval::CreateKeysPdf()
       return;
    }
 
-   std::unique_ptr<RooDataSet> chain{fChain->GetAsDataSet(SelectVars(fParameters),
-         EventRange(fNumBurnInSteps, fChain->Size()))};
+   const RooDataSet* fChainData = fChain->GetAsConstDataSet();
+   std::unique_ptr<RooDataSet> chain{
+      RooFit::makeOwningPtr<RooDataSet>(std::unique_ptr<RooAbsData>{
+         fChainData->reduce(SelectVars(fParameters), EventRange(fNumBurnInSteps, fChain->Size()))})};
+
```

Comment:
```suggestion
   std::unique_ptr<RooAbsData>{fChain->GetAsConstDataSet()->reduce(SelectVars(fParameters), EventRange(fNumBurnInSteps, fChain->Size()))};

```
Variable names starting with `f` like `fChain` are reserved for private data members. That's the ROOT convention.

Also, I would simplify this code by not casting. and then you can cast later in the single place where you use `chain`:
`*chain` -> `static_cast<RooDataSet&>(*chain)`.

Reply:
Thanks for the review!